### PR TITLE
[plugin-mobile-app] Add configuration of appium session settings

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/SessionSteps.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.List;
+
+import org.apache.commons.lang3.math.NumberUtils;
+import org.jbehave.core.annotations.When;
+import org.vividus.bdd.mobileapp.model.Entry;
+import org.vividus.selenium.IWebDriverProvider;
+
+import io.appium.java_client.ExecutesMethod;
+
+public class SessionSteps
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/SessionSteps.java:33]: there is no much sense to create a new class, I believe a new step can be put to `ApplicationSteps`

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/SessionSteps.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     */
+    @When("I change Appium session settings:$settings")
+    public void changeAppiumSettings(List<Entry> settings)
+    {
+        ExecutesMethod executesMethod = webDriverProvider.getUnwrapped(ExecutesMethod.class);
+        settings.forEach(setting ->
+        {
+            String settingValue = setting.getValue();
+            Object value = NumberUtils.isDigits(settingValue) ? Long.valueOf(settingValue) : adjustType(settingValue);
+            execute(executesMethod, new AbstractMap.SimpleEntry<>("setSettings",
```
Comment (valfirst) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/SessionSteps.java:60]: `java.util.Map#entry`

**Review by valfirst — CHANGES_REQUESTED** [review]

File: docs/modules/plugins/pages/plugin-mobile-app.adoc [hunk 3]
```diff
@@ -445,3 +445,34 @@ When I activate application with bundle identifier `com.apple.mobilesafari`
...
+==== *_Wording_*
+
+[source,gherkin]
+----
+When I change Appium session settings:$settings
+----
+
+==== *_Parameters_*
+
+. `$settings` - http://appium.io/docs/en/advanced-concepts/settings/#settings[Settings]
```
Comment (ikalinin1) [inline docs/modules/plugins/pages/plugin-mobile-app.adoc:465]: https?

File: vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/SessionSteps.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+     * <br>|name                  |value|
+     * <br>|ignoreUnimportantViews|true |
+     * <br>|snapshotMaxDepth      |100  |
+     * </code>
+     */
+    @When("I change Appium session settings:$settings")
+    public void changeAppiumSettings(List<Entry> settings)
+    {
+        ExecutesMethod executesMethod = webDriverProvider.getUnwrapped(ExecutesMethod.class);
+        settings.forEach(setting ->
```
Comment (ikalinin1) [inline vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/SessionSteps.java:56]: is there a possibility for a bulk settings update?

Comment (uarlouski) [reply vividus-plugin-mobile-app/src/main/java/org/vividus/bdd/mobileapp/steps/SessionSteps.java]: it's possible, but as per appium java client each setting update is a separate operation https://github.com/appium/java-client/blob/v7.3.0/src/main/java/io/appium/java_client/HasSettings.java

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Review by valfirst — APPROVED** [review]

**Pull request merged**